frontend/dashboard/components/RepoList/RepoList.test.tsx

```diff
@@ -32,117 +31,93 @@ const renderWithMockServices = (
+
+    expect(screen.getByText(textMock('dashboard.no_repos_result'))).toBeInTheDocument();
   });
 
-  test('should not call onSortModelChange when clicking sort button and isServerSort is false', async () => {
-    const handleSortMock = jest.fn();
+  it('should not call handleSorting when clicking sort button and isServerSort is false', async () => {
+    const user = userEvent.setup();
+    const handleSorting = jest.fn();
     renderWithMockServices({
       isServerSort: false,
-      onSortModelChange: handleSortMock,
+      onSortClick: handleSorting,
     });
-    // eslint-disable-next-line testing-library/no-node-access
-    const sortBtn = document.querySelector(
-      'button[aria-label="' + localeText.columnHeaderSortIconLabel + '"]',
-    );
-    await user.click(sortBtn);
 
-    expect(handleSortMock).not.toHaveBeenCalled();
+    await user.click(screen.getByRole('button', { name: textMock('dashboard.name') }));
+
+    expect(handleSorting).not.toHaveBeenCalled();
   });
 
-  test('should call onSortModelChange when clicking sort button and isServerSort is true', async () => {
-    const handleSortMock = jest.fn();
+  it('should call handleSorting when clicking sort button and isServerSort is true', async () => {
+    const user = userEvent.setup();
+    const handleSorting = jest.fn();
     renderWithMockServices({
       isServerSort: true,
-      onSortModelChange: handleSortMock,
+      onSortClick: handleSorting,
     });
 
-    // eslint-disable-next-line testing-library/no-node-access
-    const sortBtn = document.querySelector(
-      'button[aria-label="' + localeText.columnHeaderSortIconLabel + '"]',
-    );
-    await user.click(sortBtn);
+    await user.click(screen.getByRole('button', { name: textMock('dashboard.name') }));
 
-    expect(handleSortMock).toHaveBeenCalledWith([{ field: 'name', sort: 'asc' }], {
-      reason: undefined,
-    });
+    expect(handleSorting).toHaveBeenCalledWith('name');
   });
 
-  test('Should render GridActionsCellItem', async () => {
+  it('should call onPageChange when navigating to next page', async () => {
+    const user = userEvent.setup();
+    const onPageChange = jest.fn();
     renderWithMockServices({
-      repos: [
-        {
-          ...repository,
-          hasStarred: true,
-        },
-      ],
+      pageNumber: 0,
       isServerSort: true,
+      onPageChange,
     });
-    const unstar = await screen.findByRole('menuitem', {
-      name: textMock('dashboard.unstar', { appName: repository.name }),
-    });
-    const gridActionsCellItem = within(unstar).getByRole('img');
-    expect(gridActionsCellItem).toBeInTheDocument();
+
+    await user.click(screen.getByRole('button', { name: textMock('general.next') }));
+
+    expect(onPageChange).toHaveBeenCalledWith(1);
   });
 
-  test('should call onPageSizeChange when navigating to next / previous page', async () => {
+  it('should call onPageChange when navigating to previous page', async () => {
```

Comment:
Specifying in these test cases that the call is made with an incrementing/decrementing(?) number from the current pageNumber?

Reply:
Fixed ✅